crates/oxc_language_server/src/linter/tester.rs

```diff
@@ -0,0 +1,131 @@
+    };
+    let code_description_href = match &report.diagnostic.code_description {
+        Some(CodeDescription { href }) => href.to_string(),
+        None => "None".to_string(),
+    };
+    let message = report.diagnostic.message.clone();
+    let range = report.diagnostic.range;
+    let related_information = match &report.diagnostic.related_information {
+        Some(infos) => {
+            infos
+                .iter()
+                .enumerate()
+                .map(|(i, info)| {
+                    let mut result = String::new();
+                    result.push_str(&format!(
+                        "related_information[{}].message: {:?}",
+                        i, info.message
+                    ));
+                    // replace everything between `file://` and `oxc_language_server` with `<variable>`, to avoid
+                    // the absolute path causing snapshot test failures in different environments
+                    let mut location = info.location.uri.to_string();
+                    let start =
+                        location.find("file://").expect("file:// protocol not found in URI");
+                    let end = location
+                        .find("oxc_language_server")
+                        .expect("oxc_language_server not found in URI");
+                    location.replace_range(
+                        start + "file://".len()..end + "oxc_language_server".len(),
+                        "<variable>",
+                    );
+
+                    result.push_str(&format!(
+                        "\nrelated_information[{i}].location.uri: {location:?}",
+                    ));
+                    result.push_str(&format!(
+                        "\nrelated_information[{}].location.range: {:?}",
+                        i, info.location.range
+                    ));
+                    result
+                })
+                .collect::<Vec<_>>()
+                .join("\n")
+        }
+        None => "related_information: None".to_string(),
+    };
+    let severity = report.diagnostic.severity;
+    let source = report.diagnostic.source.clone();
+    let tags = report.diagnostic.tags.clone();
+    format!(
+        r"
+code: {code:?}
+code_description.href: {code_description_href:?}
+message: {message:?}
+range: {range:?}
+{related_information}
+severity: {severity:?}
+source: {source:?}
+tags: {tags:?}
+            "
+    )
+}
+
+/// Testing struct for the [linter server][crate::linter::server_linter::ServerLinter].
+pub struct Tester<'t> {
+    server_linter: ServerLinter,
+    snapshot_suffix: Option<&'t str>,
+}
+
+impl Tester<'_> {
+    pub fn new() -> Self {
+        Self { snapshot_suffix: None, server_linter: ServerLinter::new() }
+    }
+
+    pub fn new_with_linter(linter: Linter) -> Self {
+        Self { snapshot_suffix: None, server_linter: ServerLinter::new_with_linter(linter) }
+    }
+
+    pub fn with_snapshot_suffix(mut self, suffix: &'static str) -> Self {
```

Comment:
can trhis be removed and used file names which describe the purpose of the test

Reply:
I think some form of this is necessary, as it allows us to test on the same file, but with different configurations in the linter or editor. This is just a wrapper of what `insta` natively supports